integration/spring-data/testsuite/src/test/java/com/blazebit/persistence/spring/data/testsuite/repository/DocumentRepository.java

```diff
@@ -78,4 +79,6 @@
     Page<DocumentView> findByNameOrderById(String name, Pageable pageable, @OptionalParam("optionalParameter") String optionalParameter);
 
     List<DocumentView> findAll(Specification<Document> specification, @OptionalParam("optionalParameter") String optionalParameter);
+
+    List<DocumentView> findAll(EntityViewSettingProcessor<DocumentView> processor);
```

Comment:
Extract the `getId()` part into an interface called `IdHolderView` and add a method `<T extends IdHolderView<Long>> List<T> findAll2(EntityViewSettingProcessor<T> processor)`.

It should be possible to is like this

```
List<DocumentView> result = documentRepository.findAll2(new EntityViewSettingProcessor<DocumentView>() {
            @Override
            public EntityViewSetting<DocumentView, ?> acceptEntityViewSetting(EntityViewSetting<DocumentView, ?> setting) {
                setting.addOptionalParameter("optionalParameter", "Foo");
                return setting;
            }
});
```

Reply:
Not sure what do you mean with the `IdHolderView` part. I've updated the PR with all the other changes, I'll push this as soon as I understand better what should I change!

Reply:
I think I wanted to make sure that this kind of usage works i.e. have another findAll method that allows to return different subtypes. AFAICS the current logic would try to interpret `2` as predicate when the method is named `findAll2`. Now what do you say, would it be a good idea to allow multiple of these findAll methods or should a user just have a single one?

Reply:
The question here is how much we want to relax the Spring naming convention of method names. By default a Spring repository query method is supposed to have at least one predicate in the method name; we are now permitting no predicate embedded in the method name if:

- there is a `Specification` (any method name);
- the method name matches `^(find|read|get|query|stream)All$`.

What if we simply check the presence of the `By` keyword and skip the method name parsing if it's not present? Unfortunately Spring supports a (presumably undocumented) short syntax such as `ageGreaterThan` in place of `findByAgeGreaterThan`, so that might be a problem too.

Reply:
Now only method names matching the pattern are ignored.